daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2128,13 +2173,13 @@ func (devices *DeviceSet) removeDevice(devname string) error {
 	return err
 }
 
-func (devices *DeviceSet) cancelDeferredRemoval(info *devInfo) error {
+func (devices *DeviceSet) cancelDeferredRemovalIfNeeded(info *devInfo) error {
 	if !devices.deferredRemove {
 		return nil
 	}
 
-	logrus.Debugf("devmapper: cancelDeferredRemoval START(%s)", info.Name())
-	defer logrus.Debugf("devmapper: cancelDeferredRemoval END(%s)", info.Name())
+	logrus.Debugf("devmapper: cancelDeferredRemovalIfNeeded START(%s)", info.Name())
+	defer logrus.Debugf("devmapper: cancelDeferredRemovalIfNeeded END(%s)", info.Name())
```

Comment:
Or we can change the condition to `if devinfo != nil && devinfo.DeferredRemove != 0` and call `cancelDeferredRemoval()` inside the condition. The same way we did it inside `takeSnapshot()`. @rhvgoyal WDYT ?

https://github.com/shishir-a412ed/docker/blob/dm_task_run_failed/daemon/graphdriver/devmapper/deviceset.go#L859-l865

Shishir